src/components/article/Image.js

```diff
@@ -1,49 +1,137 @@
 'use strict'
 import _ from 'lodash'
 import classNames from 'classnames'
-import screenSize from '../../constants/screen-size'
+import { getScreenType } from '../../lib/screen-type'
 import BlockAlignmentWrapper from './BlockAlignmentWrapper'
-import MediaQuery from 'react-responsive'
 import React, { Component } from 'react'
+import ReactDOM from 'react-dom'
+import LazyLoad from 'react-lazy-load'
+import styles from './Image.scss'
 
 class Image extends Component {
   constructor(props) {
     super(props)
+
+    this.state = {
+      mounted: false,
+      screenType: 'MOBILE',
+      width: 200,
+      placeHolderOpacity: 0,
+      imageOpacity: 0
+    }
+
+    this.fitToParentWidth = this.fitToParentWidth.bind(this)
+  }
+
+  componentDidMount() {
+    this.setState({
+      mounted: true,
+      screenType: getScreenType(window.innerWidth)
+    })
+    // set state for the width of the images and listen to window resize event
+    this.fitToParentWidth()
+    window.addEventListener('resize', this.fitToParentWidth)
+  }
+
+  fitToParentWidth() {
+    const elem = ReactDOM.findDOMNode(this.refs.imageBox).parentNode
+    const width = elem.clientWidth
+    if (width !== this.state.width) {
+      this.setState({
+        width: width
+      })
+    }
+  }
+
+  _renderPlaceHoderImage(imageUrl, imgStyle) {
+    if (imageUrl) {
+      return (
+        <LazyLoad offsetTop={200}
+          onContentVisible={() => this.setState({ placeHolderOpacity: 1 })}
+          className={styles.imgPlaceHolderOuter}
+          style={ { ...imgStyle, opacity: this.state.placeHolderOpacity } }>
+          <img src={ imageUrl } className={classNames('center-block', styles.imgPlaceHolder)} style={imgStyle} />
+        </LazyLoad>
+      )
+    }
+    return null
   }
 
-  _renderFigure(imageObj) {
+  _renderFigure(imageObj, imgStyle) {
     if (imageObj) {
       return (
         <figure>
-          <img src={ imageObj.url } className={classNames('img-responsive', 'center-block')} style={{ paddingBottom: '1.5rem' }}
-          />
+          <LazyLoad offsetTop={20} onContentVisible={() => this.setState({ imageOpacity: 1 })}>
+            <img src={ imageObj.url }
+              style={ { ...imgStyle, opacity: this.state.imageOpacity } }
+              className={classNames('center-block', styles.imgAbsolute)}
+            />
+          </LazyLoad>
           { imageObj.description ? <figcaption className="image-caption" style={{ paddingTop: '1rem' }}>{ imageObj.description }</figcaption> : null}
         </figure>
       )
     }
     return null
   }
 
+  _getHeight(width, original) {
+    if (original) {
+      const oriWidth = _.get(original, 'width', width)
+      const oriHeight = _.get(original, 'height', width)
+      return width * oriHeight / oriWidth
+    }
+    return width
```

Comment:
cannot set this cause there is no content before the components are mounted, and all of the position of images is absolute.